sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/challenge_auth_policy.py

```diff
@@ -132,9 +212,9 @@ def on_challenge(self, request: PipelineRequest, response: PipelineResponse) ->
         # The tenant parsed from AD FS challenges is "adfs"; we don't actually need a tenant for AD FS authentication
         # For AD FS we skip cross-tenant authentication per https://github.com/Azure/azure-sdk-for-python/issues/28648
         if challenge.tenant_id and challenge.tenant_id.lower().endswith("adfs"):
-            self.authorize_request(request, scope)
+            self.authorize_request(request, scope, claims=challenge.claims)
         else:
-            self.authorize_request(request, scope, tenant_id=challenge.tenant_id)
+            self.authorize_request(request, scope, claims=challenge.claims, tenant_id=challenge.tenant_id)
```

Comment:
I seem to recall discussing this already but this line still makes me nervous because it appears possible for the policy to cache a CAE challenge, claims and all, and pass those claims again the next time it calls `get_token`. Can you add a test to make sure that doesn't happen?

Reply:
Good point; I expanded the `test_cae` tests to check for this (asserting that no claims are sent to a token request that comes after a successful claims challenge).

We fortunately didn't have an issue with passing past claims because we refresh the cached challenge before sending a token request, which overrides `challenge.claims` (with either `None` or a new claim). The test did expose a different bug though -- when we receive two consecutive (non-CAE) auth challenges, we were accidentally making a token request for both instead of just returning the latter 401. I updated the auth policy to avoid this by checking for claims before calling `on_challenge`.

Reply:
That's a great point, thank you for reminding me to check for token expiry! I found that while passing `enable_cae=True` to the base class means that `authorize_request` calls always get the kwarg, we still need to call `get_token` on the credential directly in some cases, and those calls weren't enabling CAE. Quick fix, and I added a test to cover the scenario.